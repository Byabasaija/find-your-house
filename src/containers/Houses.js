import PropTypes from 'prop-types';
import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { Switch, Route } from 'react-router';
import { fetchHouses } from '../api/api';
import House from '../components/House';
import UserFavorites from './UserFavorites';
import HouseDetails from './HouseDetails';

const Houses = (props) => {
  <Switch>
    <Route exact path="/house/:id" component={HouseDetails} />
    <Route exact path="/my-favorites" component={UserFavorites} />
  </Switch>;
  const { houses, fetchHouses } = props;

  useEffect(() => {
    fetchHouses();
  }, []);

  const renderHouses = () => {
    if (houses.loading === true) {
      return (
        <h2>
          Loading ...
        </h2>
      );
    } if (houses.error) {
      return (
        <h2>
          {' '}
          {houses.error}
          {' '}
        </h2>
      );
    }
    return (
      <div className="container">
        <div>
          { houses && houses.houses && houses.houses.map((ahouse) => (
            <House house={ahouse} key={ahouse.id} />))}
        </div>
      </div>
    );
  };

  return (
    <div>{ renderHouses() }</div>
  );
};

Houses.propTypes = {
  houses: PropTypes.shape({
    loading: PropTypes.bool,
    error: PropTypes.string,
    houses: PropTypes.arrayOf(PropTypes.object),
  }).isRequired,
  fetchHouses: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  houses: state.houses,
});

const mapDispatchToProps = (dispatch) => ({
  fetchHouses: () => dispatch(fetchHouses()),
});

export default connect(mapStateToProps, mapDispatchToProps)(Houses);
