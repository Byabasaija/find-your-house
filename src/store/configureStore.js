import { applyMiddleware, createStore, compose } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from '../reducers/index';

const saveToLocalStorage = (state) => {
  try {
    const serialisedState = JSON.stringify(state);
    localStorage.setItem('persistantState', serialisedState);
  } catch (e) {
    console.warn(e);
  }
};
const loadFromLocalStorage = () => {
  try {
    const serialisedState = localStorage.getItem('persistantState');
    if (serialisedState === null) return undefined;
    return JSON.parse(serialisedState);
  } catch (e) {
    console.warn(e);
    return undefined;
  }
};

const composeEnhancer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(rootReducer, loadFromLocalStorage(),
  composeEnhancer(applyMiddleware(thunk)));

store.subscribe(() => saveToLocalStorage(store.getState()));

export default store;
