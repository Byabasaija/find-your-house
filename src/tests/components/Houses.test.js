import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { PersistGate } from 'redux-persist/integration/react';
import { store, persistor } from '../../store/configureStore';
import Houses from '../../containers/Houses';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <PersistGate loading={<p>Loading</p>} persistor={persistor}>
          <BrowserRouter>
            <Houses />
          </BrowserRouter>
        </PersistGate>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
