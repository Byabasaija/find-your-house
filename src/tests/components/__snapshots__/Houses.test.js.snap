// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="container"
  >
    <nav
      className="navbar navbar-expand-lg navbar-light bg-white"
    >
      <div
        className="container"
      >
        <span
          className="navbar-brand text-dark pg-title"
          href="#"
        >
          Houses
        </span>
        <button
          aria-controls="navbarSupportedContent"
          aria-expanded="false"
          aria-label="Toggle navigation"
          className="navbar-toggler text-white"
          data-bs-target="#navbarSupportedContent"
          data-bs-toggle="collapse"
          type="button"
        >
          <i
            className="fas fa-bars text-dark"
          />
        </button>
        <div
          className="collapse navbar-collapse"
          id="navbarSupportedContent"
        >
          <ul
            className="navbar-nav me-auto mb-2 mb-lg-0"
          >
            <li
              className="nav-item"
            >
              <a
                className="nav-link active"
                href="/my-favorites"
                onClick={[Function]}
              >
                 My Favorites &gt;
              </a>
            </li>
          </ul>
          <div
            className="pb-2"
          />
          <div>
            <button
              className="btn btn-orange"
              onClick={[Function]}
              type="button"
            >
              Log out
            </button>
          </div>
        </div>
      </div>
    </nav>
    <div />
  </div>
</div>
`;
